tests/unit/test_charm.py

```diff
@@ -6,452 +6,335 @@
-            self.harness.model.unit.status,
-            BlockedStatus(
-                "'saml_target_url' cannot be specified without 'force_https' being true"
-            ),
-        )
-
-    def test_config_changed_when_no_cors(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when cors_origin configuration is empty
-        assert: it will get to blocked status waiting for it.
-        """
-        self.harness.update_config({"cors_origin": ""})
-        with self._patch_exec():
-            self.harness.container_pebble_ready("discourse")
-
-        self.assertNotEqual(
-            self.harness.charm._database.get_relation_data(),
-            None,
-            "database name should be set after relation joined",
-        )
-
-        self.assertEqual(
-            self.harness.charm._database.get_relation_data().get("POSTGRES_DB"),
-            "discourse",
-            "database name should be set after relation joined",
-        )
-
-    def test_config_changed_when_throttle_mode_invalid(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when throttle_level configuration is invalid
-        assert: it will get to blocked status waiting for a valid value to be provided.
-        """
-        self.harness.update_config({"throttle_level": "Scream"})
-
-        self.assertEqual(self.harness.model.unit.status.name, BLOCKED_STATUS)
-        self.assertTrue("none permissive strict" in self.harness.model.unit.status.message)
-
-    def test_config_changed_when_s3_and_no_bucket_invalid(self):
-        """
-        arrange: given a deployed discourse charm with all the required relations
-        act: when s3_enabled configuration is True and there's no s3_bucket
-        assert: it will get to blocked status waiting for the latter.
-        """
-        self.harness.update_config(
+@pytest.mark.parametrize(
+    "with_postgres, with_redis, with_ingress, status",
+    [
+        (False, False, False, WaitingStatus("Waiting for database relation")),
+        (False, True, False, WaitingStatus("Waiting for database relation")),
+        (True, False, False, WaitingStatus("Waiting for redis relation")),
+        (True, True, False, ActiveStatus("")),
+        (False, False, True, WaitingStatus("Waiting for database relation")),
+        (False, True, True, WaitingStatus("Waiting for database relation")),
+        (True, False, True, WaitingStatus("Waiting for redis relation")),
+        (True, True, True, ActiveStatus("")),
+    ],
+)
+def test_relations(start_harness, with_postgres, with_redis, with_ingress, status):
+    """
+    arrange: given a deployed discourse charm
+    act: when pebble ready event is triggered
+    assert: it will have the correct status depending on the relations
+    """
+    harness = start_harness(
+        with_postgres=with_postgres, with_redis=with_redis, with_ingress=with_ingress
```

Comment:
Do you think these should be fixtures rather than arguments to a function?

Reply:
I think that this is the same conversation as [here](https://github.com/canonical/discourse-k8s-operator/pull/105#discussion_r1328251271)

Reply:
Just a reminder: Fixtures usage was considered hard to follow here -> https://github.com/canonical/synapse-operator/pull/19#discussion_r1302486670